src/components/ContentWrapper.jsx

```diff
@@ -1,51 +1,65 @@
-import ContactsListDataLoader from './ContactsList/ContactsListDataLoader'
+import React, { useState } from 'react'
+import { PropTypes } from 'prop-types'
+
+import { Content } from 'cozy-ui/transpiled/react/Layout'
+
 import Header from './Header'
 import Toolbar from './Toolbar'
-import { Content } from 'cozy-ui/transpiled/react/Layout'
-import { Query } from 'cozy-client'
-
-const ContentWrapper = () => (
-  <Query
-    query={client =>
-      client
-        .all(DOCTYPE_CONTACTS)
-        .include(['accounts'])
-        .where({
-          $or: [
-            {
-              trashed: {
-                $exists: false
-              }
-            },
-            {
-              trashed: false
-            }
-          ],
-          _id: {
-            $gt: null
-          }
-        })
-        .indexFields(['_id'])
-    }
-  >
-    {({ data: contacts, fetchStatus, hasMore, fetchMore }) => {
-      return (
-        <>
-          {contacts.length >= 1 && <Header right={<Toolbar />} />}
-          <Content>
-            <ContactsListDataLoader
-              contacts={contacts}
-              fetchStatus={fetchStatus}
-              hasMore={hasMore}
-              fetchMore={fetchMore}
-            />
-          </Content>
-        </>
-      )
-    }}
-  </Query>
-)
+import ContactsList from '../components/ContactsList/ContactsList.jsx'
+import SpinnerContact from './Components/Spinner'
+import useFetchContacts from '../components/Hooks/useFetchContacts'
+import { harmonizeAndSortByFamilyNameGivenNameEmailCozyUrl } from '../helpers/contacts'
+import {
+  contactsByFamilyNameGivenNameEmailCozyUrl,
+  contactsWithoutIndexes
+} from '../helpers/queries'
+
+const ContentWrapper = ({ hasServiceBeenLaunched }) => {
+  const [isContactsWithIndexesFinished, contactsWithIndexes] = useFetchContacts(
+    contactsByFamilyNameGivenNameEmailCozyUrl
+  )
+  const [
+    isContactsWithNoIndexesFinished,
+    contactsWithNoIndexes
+  ] = useFetchContacts(contactsWithoutIndexes)
+
+  const [hasAlreadySortedContacts, setHasAlreadySortedContacts] = useState(
+    false
+  )
+  const [sortedContacts, setSortedContacts] = useState([])
+
+  const fetchesAreFinished =
+    isContactsWithIndexesFinished && isContactsWithNoIndexesFinished
+
+  if (!fetchesAreFinished) {
+    return <SpinnerContact size="xxlarge" loadingType="fetching_contacts" />
+  }
+
+  if (fetchesAreFinished && !hasAlreadySortedContacts) {
+    const reworkedContacts = hasServiceBeenLaunched
+      ? contactsWithIndexes.concat(contactsWithNoIndexes)
+      : harmonizeAndSortByFamilyNameGivenNameEmailCozyUrl(
+          contactsWithIndexes,
+          contactsWithNoIndexes
+        )
+
+    setHasAlreadySortedContacts(true)
+    setSortedContacts(reworkedContacts)
+  }
+
+  return (
+    <>
+      {sortedContacts.length >= 1 && <Header right={<Toolbar />} />}
+      <Content>
+        <ContactsList contacts={sortedContacts} />
+      </Content>
+    </>
+  )
+}
+
```

Comment:
I would like to see a few tests covering all this function. I'll be more confident on the behavior that I'm right now.

Reply:
👍  — and I wonder if it would make sense to extract the logic of this component into a standalone hook? Not only would it be easier to test, it would also mean the component only would only do something like `fetchSortedContacts`, and the implementation of `fetchSortedContacts` can change, or it can be easily reused elsewhere? Just a suggestion though.